pyfar/dsp/normalize.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+
+def normalize(input_signal, norm_type = 'time', operation = 'max', channelwise = 'max',
+              value = 'unnasigned', f = np.arange(0,22100),  fs = 44100):
+    """
+    Parameters
+    ----------
+    input_signal:
+        signal of the signal class
+        
+    norm_type:
+        'time' - normalizes time signal to 'value' 
+        'abs' - normalizes magnitude spectrum to 'value'
+        'dB' - normalizes db magnitude spectrum to 'value'
+        (default = 'time')
+
+    operation:
+        'max' - finds absolute maximum of data to normalize 
+        'mean' - find the mean of data to normalize
+        'rms' - fins the rms of data to normalize
+        (default = 'max')
+    
+    channelwise:
+        'each' - normalize each channel seperately 
+        'max' - normalize to max or operation across all channels
+        'min' - normalize to min or operation across all channels
+        'mean' - normalize to mean of operation across all channels
+        (default = 'max')
+    
+    value: 
+        normalizes to 'value' which can be a scalor or a vector 
+        with a number of elements equal to channels. Default is 0 
+        for type = 'db' and 1 otherwise
+    f:
+        two element vector specifiying upper and lower frequenzy bounds
+        for normalization or scalar specifying the centre frequency for 
+        normalization (see 'frac', default is [0,fs/2])
+    fs:
+        sampling frequency in Hz (default = 44100)
+        
+    Returns
+    --------
+    signal_normalized - normalized input signal
+    """
+    
+    # set defaults
+    
+    if value == 'unnasigned':
+        if norm_type == 'dB':
+            value = 0
+        else:
+            value = 1
+    
+    # copy input
+    
+    input_normalized = input_signal.time.copy()
+
+    # flatten
+    
+    #input_normalized =  input_normalized.flatten() # Flattening / Reshaping to a 2d array to work with possible?
+
+    # transform data to the desired domain
+    
+    if norm_type == 'time':
+         input_normalized = input_normalized
+    
+    elif norm_type == 'abs':
+        input_normalized = np.abs(np.fft.fft(input_normalized,input_normalized.ndim - 1)) 
+    
+    elif norm_type == 'dB':
+        input_normalized = np.fft.fft(input_normalized,input_normalized.ndim - 1)
+        
+    else:
+        raise ValueError(("norm _type must be 'time', 'abs' or 'dB'"))
+      
+    # get bounds for normalization
+    
+    if norm_type == 'time':
+        lim = np.array([0, input_normalized.shape[1]])
```

Comment:
You can use a tuple for this:
```suggestion
        lim = (0, signal.n_samples-1)
```

You could also use `input_normalized.shape[-1] - 1` to take the length of the last dimension. But using `Signal` methods seems more straight forward. The `-1` at the end is cleaner because it is the actual index of the last sample.